integrations/Vero/browser.js

```diff
@@ -0,0 +1,155 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable no-underscore-dangle */
+/* eslint-disable class-methods-use-this */
+import get from "get-value";
+import { NAME } from "./constants";
+import logger from "../../utils/logUtil";
+import { isDefinedAndNotNull } from "../utils/commonUtils";
+import ScriptLoader from "../ScriptLoader";
+
+class Vero {
+  constructor(config) {
+    this.apiKey = config.apiKey;
+    this.name = NAME;
+  }
+
+  init() {
+    logger.debug("===In init Vero===");
+    window._veroq = window._veroq || [];
+    ScriptLoader(null, "https://d3qxef4rp70elm.cloudfront.net/m.js", true);
+    window._veroq.push(["init", { api_key: this.apiKey }]);
+  }
+
+  isLoaded() {
+    logger.debug("in Vero isLoaded");
+    return !!window._veroq;
+  }
+
+  isReady() {
+    logger.debug("in Vero isReady");
+    return !!window._veroq && !!window._veroq.ready;
+  }
+
+  /**
+   * AddOrRemoveTags.
+   * This block handles any tag addition or removal requests.
+   * Ref - http://developers.getvero.com/?javascript#tags
+   *
+   * @param {Object} tags
+   */
+  addOrRemoveTags(tags, userId) {
+    const addTags = get(tags, "add");
+    const removeTags = get(tags, "remove");
+    window._veroq.push([
+      "tags",
+      {
+        id: userId,
+        add: addTags,
+        remove: removeTags,
+      },
+    ]);
+  }
+
+  /**
+   * Identify.
+   * Ref - https://developers.getvero.com/?javascript#users-identify
+   *
+   * @param {Identify} identify
+   */
+  identify(rudderElement) {
+    const { message } = rudderElement;
+    const { traits } = message.context || message;
+    const userId = message.userId || message.anonymousId;
+    let payload = traits;
+    if (userId) {
+      payload = { id: userId, ...payload };
+    }
+    window._veroq.push(["user", payload]);
+    const tags = message.context.integerations?.[("vero", "Vero")]?.tags;
+    if (isDefinedAndNotNull(tags)) this.addOrRemoveTags(tags, userId);
+  }
+
+  /**
+   * Track - tracks an event for a specific user
+   * for event `unsubscribe`
+   * Ref - https://developers.getvero.com/?javascript#users-unsubscribe
+   * for event `resubscribe`
+   * Ref - https://developers.getvero.com/?javascript#users-resubscribe
+   * else
+   * Ref - https://developers.getvero.com/?javascript#events-track
+   *
+   * @param {Track} track
+   */
+  track(rudderElement) {
+    logger.debug("=== In Vero track ===");
+
+    const { message } = rudderElement;
+    const { event, properties } = message;
```

Comment:
```suggestion
    const { event, properties } = rudderElement.message;
```

Reply:
Destructuring the message, as its going to use further

Reply:
You can directly destructure what ever you need directly from message

Reply:
done